src/main/java/com/iexec/core/detector/replicate/ReplicateResultUploadTimeoutDetector.java

```diff
@@ -67,15 +71,19 @@ public void detect() {
             log.info("detected replicate with resultUploadTimeout [chainTaskId:{}, replicate:{}, currentStatus:{}]",
             chainTaskId, uploadingReplicate.getWalletAddress(), uploadingReplicate.getCurrentStatus());
 
-            if (uploadingReplicate.getCurrentStatus() == ReplicateStatus.RESULT_UPLOAD_REQUESTED) {
+            if (task.getCurrentStatus() == TaskStatus.RESULT_UPLOAD_REQUESTED) {
```

Comment:
I think you'r right with this one, because *_REQUESTED is set by the pool manager, we did have a bug congrats!